src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,145 @@
-import React from 'react';
+import React, {
+  Dispatch, memo, SetStateAction, useContext, useEffect, useMemo,
+} from 'react';
+
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+import { TodosContext, TodoStatus } from '../../TodosContext';
+
+interface Props {
+  selectedUserId: number,
+  setSelectedUserId: Dispatch<SetStateAction<number>>,
+}
+
+export const TodoList: React.FC<Props> = memo(({
+  selectedUserId,
+  setSelectedUserId,
+}) => {
+  const {
+    todos,
+    todoStatus,
+    setTodoStatus,
+    todoTitle,
+    setTodoTitle,
+  } = useContext(TodosContext);
+
+  const handleChange = (event: React.ChangeEvent<
+  HTMLFormElement
+  | HTMLSelectElement
+  | HTMLInputElement
+  >) => {
+    const { value, name } = event.target;
+
+    switch (name) {
+      case 'todos-status-selector':
+        setTodoStatus(value);
+        break;
+
+      case 'todos-title-input':
+        setTodoTitle(value);
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  const prepareTodos = (preparedTodos: Todo[]) => (
```

Comment:
```suggestion
  const prepareTodos = (todosToPrepare: Todo[]) => (
```
Якось так, бо якшо вони вже prepared, то нашо нам функція prepare